pkg/tuple/tuple_test.go

```diff
@@ -468,3 +468,75 @@ func TestTypedPublicWildcard(t *testing.T) {
 	require.Equal(t, "group:*", TypedPublicWildcard("group"))
 	require.Equal(t, ":*", TypedPublicWildcard("")) // Does not panic
 }
+
+func TestParseTupleString(t *testing.T) {
+	tests := []struct {
+		name        string
+		str         string
+		tuple       *openfgav1.TupleKey
+		expectedErr bool
+	}{
+		{
+			name:  "well_formed_user_object",
+			str:   "document:1#viewer@user:jon",
+			tuple: NewTupleKey("document:1", "viewer", "user:jon"),
+		},
+		{
+			name:  "well_formed_user_userset",
+			str:   "document:1#viewer@group:eng#member",
+			tuple: NewTupleKey("document:1", "viewer", "group:eng#member"),
+		},
+		{
+			name:        "missing_user_field",
+			str:         "document:1#viewer",
+			expectedErr: true,
+		},
+		{
+			name:        "missing_relation_separator",
+			str:         "document:1viewer@user:jon",
+			expectedErr: true,
+		},
+		{
+			name:        "missing_object_id",
+			str:         "document#viewer@user:jon",
+			expectedErr: true,
+		},
+		{
+			name:        "malformed_object_type",
+			str:         "docu#ment:1#viewer@user:jon",
+			expectedErr: true,
+		},
+		{
+			name:        "malformed_object_id",
+			str:         "document:#1#viewer@user:jon",
+			expectedErr: true,
+		},
+		{
+			name:        "malformed_relation",
+			str:         "document:1#vie:wer@user:jon",
+			expectedErr: true,
+		},
+		{
+			name:        "malformed_user_type",
+			str:         "document:1#viewer@use#r:jon",
+			expectedErr: true,
+		},
+		{
+			name:        "malformed_user_userset",
+			str:         "document:1#viewer@group:e:eng#member",
+			expectedErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tuple, err := ParseTupleString(test.str)
+			require.Equal(t, test.expectedErr, err != nil)
+
+			if !test.expectedErr {
+				require.NotNil(t, tuple)
+				require.Equal(t, test.tuple, tuple)
+			}
```

Comment:
```suggestion
if test.expectedErr != nil {
	require.ErrorIs(t, actualError, test.expectedErr)
} else {
	require.NotNil(t, tuple)
	require.Equal(t, test.tuple, tuple)
}
```

Reply:
We're not returning sentinel errors, so `ErrorIs` doesn't work here. The intent was not to test the error message as that is subject to change, but to assert that an error is raised. So that was the intent of the bool for `test.expectedErr` vs error type.